File: src/main/java/com/online/model/Item.java
package com.online.model;

import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.util.List;

@Entity
@Table(name = "items")
@Data
@NoArgsConstructor
public class Item {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(nullable = false, updatable = false)
    private Long id;
    private String title;
    private String manufacturer;
    private Double price;
    private String category;
    private String description;
    private Integer warranty;
    private String image;
    private Integer amountInStock;
    private Double rating;
    private Integer ratingsAmount;

    @OneToMany(fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    @JoinColumn(name = "itemId")
    private List<Comment> comments;

}

File: src/main/online-store/src/app/models/item/item.ts
import {CommentModel} from "./comment.model";

export class Item {

  id: number;
  title:string;
  manufacturer:string;
  price:number;
  category:string;
  description:string;
  image:string;
  amount: number;
  rating: number;
  ratingsAmount: number;
  comments: CommentModel[];
}
